srcpkgs/rpi-kernel/template

```diff
@@ -52,31 +51,109 @@ if [ "$CROSS_BUILD" ]; then
 	_cross="CROSS_COMPILE=${XBPS_CROSS_TRIPLET}-"
 fi
 
+# handle all raspberry pi flavors
+case "$XBPS_TARGET_MACHINE" in
+	armv6*) _flavors=rpi1 ;;
+	armv7*) _flavors=rpi2 ;;
+	aarch64*) _flavors="rpi3 rpi4" ;;
+esac
+
+subpackages="rpi-kernel-headers"
+for _f in $_flavors; do
+	subpackages+=" ${_f}-kernel-headers ${_f}-kernel"
+done
+
+# support legacy installations. don't drop before 2021-12 !!
+# make the meta packages depend on the new flavor packages
+case "$XBPS_TARGET_MACHINE" in
+	armv6*) depends=rpi1-kernel ;;
+	armv7*) depends=rpi2-kernel ;;
+	aarch64*) depends=rpi3-kernel ;;
```

Comment:
These dependencies are also set for the flavor packages, causing for example `rpi4-kernel` to depend on `rpi3-kernel`, since `rpi4-kernel` is built for `aarch64`. This is definitely unwanted, and causes problems with `./mkplatformfs.sh rpi4 ...` when it attempts to install both `rpi3-kernel` and `rpi4-kernel`, as the files will conflict.

Reply:
oh right. will fix it. transition handling is not ideal yet

Reply:
i cannot confirm. `depends=` is not inherited by subpakages. the toplevel declaration you refereced only affects the meta packages, see below (notice `run_depends`)
```
$ XBPS_TARGET_ARCH=aarch64-musl xbps-query --repository=$HOME/src/void-packages/hostdir/binpkgs -M -R rpi4-kernel                                                                                       
architecture: aarch64-musl
filename-sha256: 422a8449148b60cf6dd53f80f61f2b556b4bb505cc6588e6c5877721e6273054
filename-size: 28MB
homepage: http://www.kernel.org
installed_size: 39MB
license: GPL-2.0-only
maintainer: Piraty <piraty1@inbox.ru>
pkgname: rpi4-kernel
pkgver: rpi4-kernel-5.4.68_1
repository: /home/piraty/src/void-packages/hostdir/binpkgs
short_desc: Linux kernel for Raspberry Pi 4 (5.4 [git e82816d])

$ XBPS_TARGET_ARCH=aarch64-musl xbps-query --repository=$HOME/src/void-packages/hostdir/binpkgs -M -R rpi-kernel
architecture: aarch64-musl
filename-sha256: 0686ecf163d116935cac4c2f744ff3d00e2baf023ff952cd4c4c143ba4752a24
filename-size: 529B
homepage: http://www.kernel.org
installed_size: 0B
license: GPL-2.0-only
maintainer: Piraty <piraty1@inbox.ru>
pkgname: rpi-kernel
pkgver: rpi-kernel-5.4.68_1
repository: /home/piraty/src/void-packages/hostdir/binpkgs
run_depends:
        rpi3-kernel>=0
short_desc: Linux kernel for Raspberry Pi (5.4 series [git e82816d])
```

Reply:
Yes, I get the same result that you do. However this is still somehow a problem, even if this isn't the source. When I run `./xbps-src -a aarch64 pkg rpi4-kernel`, for some reason. it also builds `rpi-kernel`, `rpi-kernel-headers`, and transitively, `rpi3-kernel` and `rpi3-kernel-headers`. And then, for some reason, `./mkplatformfs.sh rpi4 ...` attempts to install them.

```
$ ls hostdir/binpkgs/rpi-kernel-flavor-subpackage/
aarch64-repodata                          rpi3-kernel-headers-5.4.68_1.aarch64.xbps
rpi-kernel-5.4.68_1.aarch64.xbps          rpi4-kernel-5.4.68_1.aarch64.xbps
rpi-kernel-headers-5.4.68_1.aarch64.xbps  rpi4-kernel-headers-5.4.68_1.aarch64.xbps
rpi3-kernel-5.4.68_1.aarch64.xbps
```
see also: the [build log](https://termbin.com/9he0z) for `./xbps-src -a aarch64 pkg rpi4-kernel`, which builds both rpi3-kernel and rpi4-kernel.

Reply:
@agausmann the behavior you described for building subpackages is expected. You build a source package, and all of its subpackages get built. If there are issues with how mkplatformfs handles things, then we can iron it out in the PR over there.